File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findSecondMinimumValue = function(root) {
    let min = root.val
    let secondMin = Infinity
    function dfs(node) {
        if(!node) return
        if(node.val>min&&node.val<secondMin){
            secondMin = node.val
        }
        dfs(node.left)
        dfs(node.right)
    }
    dfs(root)
    if(secondMin==Infinity){
        return -1
    } else {
        return secondMin
    }
};

File: 0671-second-minimum-node-in-a-binary-tree/0671-second-minimum-node-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findSecondMinimumValue(self, root: Optional[TreeNode]) -> int:
        min_val = root.val
        second_min = float('inf')

        def dfs(node):
            nonlocal second_min
            if not node:
                return
            if min_val < node.val < second_min:
                second_min = node.val
            dfs(node.left)
            dfs(node.right)

        dfs(root)
        return -1 if second_min == float('inf') else second_min